src/ser/error.rs

```diff
@@ -17,6 +17,9 @@ pub enum Error {
         key: Bson,
     },
 
+    /// Attempted to serialize a sub-millisecond precision datetime.
+    SubMillisecondPrecisionDateTime(crate::DateTime),
```

Comment:
Does this error name seem okay? it's a little verbose but I couldn't come up with anything much better.

Reply:
agree it's fine. might be worth expanding the comment a little to clarify _why_ that's an error though, e.g. adding "...which BSON does not support" or something like that?

Reply:
Good idea, added.